### PR TITLE
Voice recognition cleanup based on Mantis #2373 patch.

### PR DESCRIPTION
**Pull request by MageKing17**
* Frees `cpAudio` pointer in `VOICEREC_deinit()`
* Validates that the ship/wing can actually take orders, and validates that it can take the specific order in question before trying to send it.
* Reports the confidence of the voice match if `DEBUG_ON` is set.
* Various commenting changes based on the patch.

Not included: using confidence to discard voice matches that fall below a certain confidence level. I haven't noticed a tendency towards spurious matches, and I didn't want to introduce this without additional data, especially when it might cause problems for people with different accents (and ESPECIALLY without a setting to tweak the threshold value, which the original patch did not have; it was hardcoded at 90%).

[Mantis link](http://scp.indiegames.us/mantis/view.php?id=2373)

### CODE REVIEW

File: code/sound/voicerec.cpp [hunk 1]
```diff
@@ -445,10 +457,11 @@ void VOICEREC_execute_command(ISpPhrase *pPhrase, HWND hWnd)
 
 				break;
 			}
+			// phrases for transferring shield energy to different locations
 			case VID_shields:
 			{
 				int action = pElements->pProperties->vValue.ulVal;
-				mprintf(("Shield Transfer %d \n", action));
+				//mprintf(("Shield Transfer %d \n", action));
```
Comment (asarium) [inline code/sound/voicerec.cpp]: Maybe completely remove this?

File: code/sound/voicerec.cpp [hunk 2]
```diff
@@ -462,12 +475,14 @@ void VOICEREC_execute_command(ISpPhrase *pPhrase, HWND hWnd)
...
 			}
+			// basic cheat, phrase as a value equivalent to the defines in ControlConfig/ControlsConfig.h
+			// it just calls the button_function with this value
 			case VID_speed:
 			case VID_targeting:
 			case VID_other:
 			{
 				int action = pElements->pProperties->vValue.ulVal;
-				mprintf(("Targeting/speed %d \n", action));
+				//mprintf(("Targeting/speed %d \n", action));
```
Comment (asarium) [inline code/sound/voicerec.cpp]: This as well.

**Comment by asarium** [comment]
LGTM, the two commented lines could be removed. They don't add much in terms of documentation and don't contain code that could be useful later.

**Comment by MageKing17** [comment]
Commit ammended to remove all the commented-out code (most of it was similarly useless and even if somebody wanted it for some reason, that's what version control is for) and also I changed my mind about the hudsquadmsg.h changes, so give this another quick lookover.

**Comment by asarium** [comment]
Looked at it again, still LGTM.

**Pull request merged**